### PR TITLE
91 commit func

### PR DESCRIPTION
**Pull request by ddd-mtl**
First working version of Commit function.
There is a lot of data handling code in the wasm code. I don't know if there is a way around that.
Also I'm using the same memory buffer as the one used for the result of the Zome call.

### CODE REVIEW

File: core_api/wasm-test/commit/src/lib.rs [hunk 1]
```diff
@@ -70,25 +60,25 @@ fn hc_commit(data: *mut c_char, entry_type_name: &str, entry_content : &str)
...
-// Convert a data struct into json memory buffer
-fn serialize<T: Serialize>(data: *mut c_char, internal: T) -> i32 {
-    let json = serde_json::to_string(&internal).unwrap();
-    let bytes = json.as_bytes();
-    let len = bytes.len();
-    let mem = unsafe { slice::from_raw_parts_mut(data, len) };
+// Write a data struct into a memory buffer as json string
+fn serialize<T: Serialize>(ptr_data: *mut c_char, internal: T) -> i32 {
+    let json_str = serde_json::to_string(&internal).unwrap();
+    let json_bytes = json_str.as_bytes();
```
Comment (sphinxc0re) [inline core_api/wasm-test/commit/src/lib.rs]: https://docs.serde.rs/serde_json/fn.to_vec.html
This should do the trick

Comment (ddd-mtl) [reply core_api/wasm-test/commit/src/lib.rs]: done

**Comment by lucksus** [comment]
Besides my comment on ```chain.puhs(entry);``` it looks good though.

**Comment by ddd-mtl** [comment]
I added Julian's change request so for me that counted as 2nd approval.

**Pull request merged**